identity_storage/src/secret_manager_wrapper.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright 2020-2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use iota_sdk::client::secret::stronghold::StrongholdSecretManager;
+use iota_sdk::client::secret::SecretManager;
+use iota_stronghold::Stronghold;
+use std::sync::Arc;
+use tokio::sync::MutexGuard;
+
+/// Wrapper around `SecretManager` that implements the storage interfaces.
+#[derive(Clone)]
+pub struct SecretManagerWrapper(Arc<SecretManager>);
```

Comment:
This is a type that (conceptually) wraps `StrongholdSecretManager`, not `SecretManager` in general, so this should be reflected in the type's name. I would suggest `StrongholdStorage` or `StrongholdSecretStorage`, or something along those lines. Don't forget to rename the file too.